client/client_inventory_test.go

```diff
@@ -35,12 +34,14 @@ func TestInventoryClient(t *testing.T) {
 	}
 
 	// Test server that always responds with 200 code, and specific payload
-	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	ts := startTestHTTPS(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(responder.httpStatus)
 
 		responder.recdata, _ = ioutil.ReadAll(r.Body)
 		responder.path = r.URL.Path
-	}))
+	}),
+		localhostCert,
+		localhostKey)
```

Comment:
For all these tests, we use a cert loaded from string (from `tests_start_https.go`) and also as a filename parameter for `NewApiClient`. I wonder if we can eliminate the duplication by making `startTestHTTPS` read the certs from the disk.

Reply:
I think I am missing something, as some tests do use different certs for `startTestHTTPS` and `NewApiClient`.

Reply:
yes, thats the main idea. different certs combinations trigger different errors handled in `dialOpenSSL`.

Reply:
Okey.

But still the strings at `startTestHTTPS` could be loaded from the files instead of duplicating the certs.

Reply:
I will do what I can!

Reply:
ok, I took another look, Lluis maybe we can keep it like that? every certificate in `https_server_test.go` (was: `tests_start_https.go`) is different, I mean all the variables:
```go
var localhostCertUnknown
var localhostKeyUnknown
var localhostCertExpired
var localhostKeyExpired
var localhostCert
var localhostKey
var localhostCertShortEEKey
var localhostKeyShortEEKey
```
hold different data, to trigger different errors in verification. if you insist I will load them form files, but you will have to convince me :) what are the benefits here?

Reply:
I meant that, for example, `var localhostCertExpired` is the same content as file `client/server.expired.crt`.  So for these (variables that are duplicated in files) we could just read the files.

It is not a big deal anyway, leave it as-is if you like it more.